src/Spectre.libClassifier/OpenCvDataset.cpp

```diff
@@ -19,14 +19,25 @@ limitations under the License.
 
 #include "Spectre.libException/InconsistentArgumentSizesException.h"
 #include "Spectre.libException/OutOfRangeException.h"
+#include "Spectre.libException/EmptyOpenCvDatasetException.h"
 #include "OpenCvDataset.h"
 
 namespace Spectre::libClassifier {
 
 const int ColumnMatrixWidth = 1;
 
+bool throwOnEmpty(_int64 size)
```

Comment:
This function may accept size and just pass it with / without exception. We're calling both `data.size()` and `labels.size()` to evaluate parameters of `cv::Mat` constructor - when we throw at this point, it's fine.